Mathlib/Data/Finset/Pi.lean

```diff
@@ -194,6 +194,12 @@ lemma restrict_preimage [DecidablePred (· ∈ s)] (t : (i : s) → Set (π i))
   ext
   simp_all
 
+lemma restrict_preimage' [DecidableEq ι] {I : Set ι}
+    [DecidablePred (· ∈ I)] (s : Finset I) (u : (i : I) → Set (π i)) :
+    I.restrict ⁻¹' Set.pi s u =
+      Set.pi (s.image Subtype.val) (fun i ↦ if h : i ∈ I then u ⟨i, h⟩ else .univ) := by
+  grind
```

Comment:
I think I would rename the first of these two `restrict_preimage_univ` and let the latter take its name.